framework/src/main/java/org/checkerframework/framework/util/defaults/QualifierDefaults.java

```diff
@@ -352,7 +352,18 @@ private boolean conflictsWithExistingDefaults(
      * @param type the type to annotate
      */
     public void annotate(Element elt, AnnotatedTypeMirror type) {
```

Comment:
I'm wondering whether it would be nicer to directly expose the boolean in the public API.
This would be a breaking change, but would force people to adapt to this changed logic. At the moments existing clients would silently not override all the right methods and get strange behavior.
It would also keep this class a bit more manageable.